tests/unit/s2n_array_test.c

```diff
@@ -87,6 +87,37 @@ int main(int argc, char **argv)
     EXPECT_EQUAL(array->element_size, element_size);
     EXPECT_SUCCESS(memcmp(array->elements, mem.data, num_of_elements * element_size));
 
+    /* Insert element at given index */
+    struct array_element *insert_element = s2n_array_insert(array, 16);
+    insert_element->first = 20;
+    insert_element->second = 'a' + 20;;
+
+    /* Validate array parameters */
+    EXPECT_EQUAL(array->capacity, 32);
+    EXPECT_EQUAL(array->num_of_elements, 18);
+
+    /* Get the inserted element */
+    struct array_element *inserted_element = s2n_array_get(array, 16);
+    EXPECT_EQUAL(inserted_element->first, 20);
```

Comment:
rather than the constants, should you be using:

insert_element->first and second on these two lines?

Reply:
Done.